src/mcts/search.cc

```diff
@@ -860,6 +863,11 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
     int possible_moves = 0;
     const float fpu = GetFpu(params_, node, is_root_node);
     for (auto child : node->Edges()) {
```

Comment:
It crashes when it happens that all moves have `GetN() > 0` (and `possible_moves == 0`).
Then during the next iteration `best_edge` is used, but it doesn't point to anything useful.

Reply:
So, maybe workaround would be to pass the collision event max flag to 1 (like it used to be?)

Reply:
Seems collisions event max of 1 was not sufficient.  But out of order eval being disabled is (or maybe is in combination with setting collisions max to 1).

Reply:
actually none of that was sufficient - so I made some more fixes.
Along the way I think I found and fixed a performance issue.

But still needs threads=1 to be correct I think, otherwise there is a chance that the least desirable child node won't be evaluated before the result is returned.  Could be fixed by changing the flag to only be set if there are 0 possible moves, rather than 1 or less.